apm-agent-plugins/apm-asynchttpclient-plugin/src/test/java/co/elastic/apm/agent/asynchttpclient/AsyncHttpClientInstrumentationTest.java

```diff
@@ -49,11 +47,35 @@ public AsyncHttpClientInstrumentationTest(RequestExecutor requestExecutor) {
         this.requestExecutor = requestExecutor;
     }
 
+    public static AsyncHandler<Response> customAsyncHandler = new AsyncCompletionHandler<Response>() {
+        @Override
+        public State onStatusReceived(HttpResponseStatus responseStatus) {
+            assertThat(tracer.currentTransaction()).isNotNull();
+            assertThat(tracer.currentTransaction().isExit()).isTrue();
+            return State.CONTINUE;
+        }
+
+        @Override
+        public void onThrowable(Throwable t) {
+            assertThat(tracer.currentTransaction()).isNotNull();
+            assertThat(tracer.currentTransaction().isExit()).isTrue();
+        }
+
+        @Override
+        public Response onCompleted(Response response) {
+            assertThat(tracer.currentTransaction()).isNotNull();
```

Comment:
To double-check, do the tests fail when doing `isNull()`?

Reply:
Ill double-check all PR requirements now and make sure the tests make sense :)

Reply:
Mmm, the changes make the existing tests fail (even when I remove the newly added `customAsyncHandler`. 

```
testHttpCall[3]  Time elapsed: 0.519 sec  <<< ERROR!
org.awaitility.core.ConditionTimeoutException:
Assertion condition defined as a lambda expression in co.elastic.apm.agent.MockReporter
Expecting actual not to be empty within 500 milliseconds.
```

Seems to me that is a local problem linked to my laptop. Will try to figure out why it is doing that. Any chance you have a docker-image able to run the tests?

Reply:
The tests are failing on CI for the same reason. Not sure what causes that 🤔

Reply:
Oh wow, interesting. Very strange that the new instrumentations make the current tests time-out. Will have a look later today when I have a bit more time :)

Reply:
Fixed it :)
Trying to get the new tests to work now as they currently only pass on the assertion of `isNull()`.

Could it be that activating the span is not enough and that the tracer needs to set the transaction to active instead?

Reply:
hm, strange.. No, activating the span is enough. You don't have to activate the transaction on this thread.

Reply:
I tried to figure it out but tbh, I have no idea.
Only thing I could think of was that Im using a `new AsyncCompletionHandler<Response>` in the test and somehow the methods of this subclass don't get matched properly.

Reply:
@felixbarny I've played around with it a bit more:
```
@Advice.OnMethodEnter(suppress = Throwable.class)
        private static void onMethodEnter(@Advice.This AsyncHandler<?> asyncHandler, @Advice.Local("span") Span span) {
            span = handlerSpanMap.remove(asyncHandler);
            if (span != null) {
                span.activate();
                System.out.println(tracer.currentTransaction());
                System.out.println("=======================");
            }
        }
```

This prints out:
```
null
=======================
```

So it seems that activating the span does not mean I can do `assertThat(tracer.currentTransaction()).isNull();` in the test.

Reply:
Yes, that's true. You should assert that `tracer.getActive()` is not null instead. I agree that if `tracer.getActive()` returns non-null, `tracer.getTranaction()` should do, too. I have an idea how to accomplish that. In the meantime, please assert on `tracer.getActive()`.

Reply:
See also #1174

Reply:
Cool, thanks :)
Ive applied the changes and the tests now pass correctly!